src/lib/y2storage/fake_device_factory.rb

```diff
@@ -609,13 +609,15 @@ def create_lvm_pv(parent, args)
     def create_btrfs(parent, args)
       log.info("#{__method__}( #{parent}, #{args} )")
       default_subvolume = args["default_subvolume"]
-      if default_subvolume
-        blk_device = Y2Storage::BlkDevice.find_by_name(@devicegraph, parent)
+      if default_subvolume && !default_subvolume.empty?
+        blk_device = BlkDevice.find_by_name(devicegraph, parent)
         filesystem = blk_device.filesystem
         if !filesystem || !filesystem.type.is?(:btrfs)
           raise HierarchyError, "No btrfs on #{parent}"
         end
-        filesystem.create_btrfs_subvolume(@default_subvolume)
+        btrfs = Storage.to_btrfs(filesystem)
```

Comment:
This is still combining the `Storage` and `Y2Storage` namespaces and downcasting stuff in the former. This should not happen in the YAML-reader. Introducing this is against https://trello.com/c/1gvNrbFQ/546-8-wrapper-based-api  

Maybe there is a bug in the Ruby wrapper, maybe a problem in another place, but this doesn't seem like the way to go.